Quadriga/src/test/java/edu/asu/spring/quadriga/service/impl/projectblog/ProjectBlogEntryManagerTest.java

```diff
@@ -0,0 +1,176 @@
+
+        // Calling the method to test
+        List<IProjectBlogEntry> projectBlogEntryList = projectBlogEntryManagerUnderTest
+                .getProjectBlogEntryList(DUMMY_PROJECT_ID, LIMIT);
+
+        // Asserting the details
+        assertProjectBlogEntryList(projectBlogEntryList);
+    }
+
+    private void assertProjectBlogEntryList(List<IProjectBlogEntry> projectBlogEntryList) {
+
+        // Assert that exactly 2 entries are fetched
+        assertEquals(projectBlogEntryList.size(), 2);
+
+        // Assert the details of first blog fetched only when atleast one entry
+        // is fetched
+        // If there are fewer entries than expected, first assert will fail and
+        // NullPointerException is avoided
+        if (projectBlogEntryList.size() >= 1) {
+            assertEquals(projectBlogEntryList.get(0).getProjectBlogEntryId(), BLGE_ID_1);
+            assertEquals(projectBlogEntryList.get(0).getTitle(), TITLE_1);
+            assertEquals(projectBlogEntryList.get(0).getDescription(), DESCRIPTION_1);
+        }
+        if (projectBlogEntryList.size() >= 2) {
+            assertEquals(projectBlogEntryList.get(1).getProjectBlogEntryId(), BLGE_ID_2);
+            assertEquals(projectBlogEntryList.get(1).getTitle(), TITLE_2);
+            assertEquals(projectBlogEntryList.get(1).getDescription(), DESCRIPTION_2);
+        }
+    }
+
+    private ProjectBlogEntryDTO createProjectBlogEntryDTO(String id, String title, String description) {
+        ProjectBlogEntryDTO projectBlogEntryDTO = new ProjectBlogEntryDTO();
+        projectBlogEntryDTO.setProjectBlogEntryId(id);
+        projectBlogEntryDTO.setTitle(title);
+        projectBlogEntryDTO.setDescription(description);
+        return projectBlogEntryDTO;
+    }
+
+    /**
+     * tests
+     * {@link IProjectBlogEntryManager#addNewProjectBlogEntry(IProjectBlogEntry)}
+     * method responsible for adding a project blog entry into database
+     * 
+     * @throws QuadrigaStorageException
+     */
+    @Test
+    public void addNewProjectBlogEntryTest() throws QuadrigaStorageException {
+
+        // Creating dummy project blog entry
+        IProjectBlogEntry projectBlogEntry = new ProjectBlogEntry();
+        projectBlogEntry.setTitle(TITLE_NEW);
+        projectBlogEntry.setDescription(DESCRIPTION_NEW);
+
+        Mockito.when(projectBlogEntryDAO.generateUniqueID()).thenReturn(UNIQUE_BLGE_ID);
+
+        // Mocked ProjectBlogEntryDTO instance
+        ProjectBlogEntryDTO projectBlogEntryDTO = createProjectBlogEntryDTO(UNIQUE_BLGE_ID, TITLE_NEW, DESCRIPTION_NEW);
+
+        Mockito.when(projectBlogEntryDTOMapper.getProjectBlogEntryDTO(projectBlogEntry))
+                .thenReturn(projectBlogEntryDTO);
+
+        // Calling the method to test
+        projectBlogEntryManagerUnderTest.addNewProjectBlogEntry(projectBlogEntry);
+
+        // Verifying that saveNewDTO method is called
+        verify(projectBlogEntryDAO).saveNewDTO(projectBlogEntryDTO);
```

Comment:
you also want to make sure that the projectBlogEntry that is passed to your mapper has a date and and id.